storage/gc_queue.go

```diff
@@ -236,27 +236,20 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 	wg.Wait()
 
 	// Resolve all intents.
-	// TODO(spencer): use a batch here when available.
+	var intents []proto.Intent
 	for id, txn := range txnMap {
 		if txn.Status != proto.PENDING {
-			// The transaction was successfully pushed, so resolve the intents.
-			for _, key := range intentMap[id] {
-				resolveArgs := &proto.ResolveIntentRequest{
-					RequestHeader: proto.RequestHeader{
-						Timestamp: now,
-						Key:       key,
-						User:      security.RootUser,
-						Txn:       txn,
-					},
-				}
-				if _, err := repl.AddCmd(repl.context(), resolveArgs); err != nil {
-					log.Warningf("resolve of key %q failed: %s", key, err)
-					updateOldestIntent(txn.OrigTimestamp.WallTime)
-				}
+			for _, intent := range intentMap[id] {
+				intent.Txn = *txn
+				intents = append(intents, intent)
 			}
 		}
 	}
 
+	if len(intents) > 0 {
+		repl.resolveIntents(repl.context(), intents)
```

Comment:
as per our discussion (I will amend #2062 to reflect it) this is hopefully ok: as long as we don't delete transaction table entries, it's perfectly fine if these fail (though, yeah, they never should).
Only Txn entries that don't have intents registered to them will be GCed. The way an intent gets unregistered is that `ResolveIntent` returns successfully and a follow-up RPC updates the txn record (which will also delete it right away if the update empties the list, largely obviating the need for the GC queue to do anything; I doubt it'll actually do any work on the `Txn` table in practice).